stack_data/formatting.py

```diff
@@ -168,7 +179,7 @@ def format_line(self, line: Line) -> str:
             result += " "
 
         if self.show_linenos:
-            result += "{:4} | ".format(line.lineno)
+            result += self.line_number_format_string.format(line.lineno)
```

Comment:
Why this new option? It seems unrelated to the rest of the PR. And shouldn't it be used in `format_blank_lines_linenumbers`?

Reply:
I added it as the indentation was fixed and I might want a larger value in friendly_traceback; it is the only hard-coded value you had in stack_data, while all the others could be changed by the users. And, yes, I overlooked the fact that it should have been used in `format_blank_lines_linenumbers` as well. I will need to do this.